calico.go

```diff
@@ -0,0 +1,302 @@
+			// Otherwise, incoming traffic is only allowed from profiles with the same tag.
+			k8sInboundRule := []libcalico.Rule{libcalico.Rule{Action: "allow"}}
+			tagInboundRule := []libcalico.Rule{libcalico.Rule{Action: "allow", SrcTag: conf.Name}}
+			fmt.Fprintf(os.Stderr, "Calico CNI creating profile. profile=%s\n", conf.Name)
+
+			var inboundRule []libcalico.Rule
+			if runningUnderK8s {
+				inboundRule = k8sInboundRule
+			} else {
+				inboundRule = tagInboundRule
+			}
+
+			profile := libcalico.Profile{
+				ID: conf.Name,
+				Rules: libcalico.Rules{
+					Inbound:  inboundRule,
+					Outbound: []libcalico.Rule{libcalico.Rule{Action: "allow"}}},
+				Tags: []string{conf.Name}}
+			if err := profile.Write(etcd); err != nil {
+				return err
+			}
+		}
+	}
+
+	return result.Print()
+}
+
+func cmdDel(args *skel.CmdArgs) error {
+	n := NetConf{}
+	if err := json.Unmarshal(args.StdinData, &n); err != nil {
+		return fmt.Errorf("failed to load netconf: %v", err)
+	}
+
+	if err := AddIgnoreUnknownArgs(); err != nil {
+		return err
+	}
+
+	// Always try to release the address
+	if err := ipam.ExecDel(n.IPAM.Type, args.StdinData); err != nil {
+		return err
+	}
+
+	// Always try to clean up the workload/endpoint.
+	// First determine if running under k8s to get the right workload and orchestrator IDs
+	k8sArgs := K8sArgs{}
+	if args.Args != "" {
+		err := LoadArgs(args.Args, &k8sArgs)
+		if err != nil {
+			return err
+		}
+	}
+	var orchestratorID, workloadID string
+
+	runningUnderK8s := string(k8sArgs.K8S_POD_NAMESPACE) != "" && string(k8sArgs.K8S_POD_NAME) != ""
+	if runningUnderK8s {
+		workloadID = fmt.Sprintf("%s.%s", k8sArgs.K8S_POD_NAMESPACE, k8sArgs.K8S_POD_NAME)
+		orchestratorID = "k8s"
+	} else {
+		workloadID = args.ContainerID
+		orchestratorID = "cni"
+	}
+
+	// Actually remove the workload
+	etcd, err := libcalico.GetKeysAPI(n.EtcdAuthority, n.EtcdEndpoints)
+	if err != nil {
+		return err
+	}
+	workload := libcalico.Workload{
+		Hostname:       hostname,
+		OrchestratorID: orchestratorID,
+		WorkloadID:     workloadID}
+	if err := workload.Delete(etcd); err != nil {
+		return err
+	}
+
+	// Only try to delete the device if a namespace was passed in
+	if args.Netns != "" {
+		err := ns.WithNetNSPath(args.Netns, func(_ ns.NetNS) error {
+			var err error
+			_, err = ip.DelLinkByNameAddr(args.IfName, netlink.FAMILY_V4)
+			return err
+		})
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// VERSION is filled out during the build process (using git describe output)
+var VERSION string
+
+func main() {
+	// Display the version on "-v", otherwise just delagate to the skel code.
+	// Use a new flag set so as not to conflict with existing libaries which use "flag"
+	flagSet := flag.NewFlagSet("Calico", flag.ExitOnError)
+
+	version := flagSet.Bool("v", false, "Display version")
+	err := flagSet.Parse(os.Args[1:])
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+	if *version {
+		fmt.Println(VERSION)
```

Comment:
This doesn't match the output format of the Python code (a dictionary).  I remember at least one user was checking the output of `-v` programmatically.  I think this is an incompatible change.